pkg/kubelet/kubelet.go

```diff
@@ -368,21 +364,19 @@ func NewMainKubelet(
 	}
 	klet.containerManager = containerManager
 
-	go util.Until(klet.syncNetworkStatus, 30*time.Second, util.NeverStop)
-	if klet.kubeClient != nil {
-		// Start syncing node status immediately, this may set up things the runtime needs to run.
```

Comment:
Tried to add some information. After this refactor, the start up requirements should be much easier to wrap your head around.